tests/sentry/event_manager/test_event_manager.py

```diff
@@ -120,6 +120,51 @@ def make_release_event(self, release_name: str, project_id: int) -> Event:
 
 
 class EventManagerTest(TestCase, SnubaTestCase, EventManagerTestMixin, PerformanceIssueTestCase):
+    def test_group_in_deletion_in_progress_should_not_create_new_group(self) -> None:
+        data = {"timestamp": before_now(minutes=1).isoformat()}
+
+        event = self.store_event(data=data, project_id=self.project.id)
+        assert event.group_id is not None
+        group = event.group
+        first_event_hash = event.get_primary_hash()
+
+        # New events will still be associated with the group in deletion in progress
+        group.update(status=GroupStatus.DELETION_IN_PROGRESS, substatus=None)
+        group.save()
+        new_event = self.store_event(data=data, project_id=self.project.id)
+        # The new event has the same hash as the first event
+        assert new_event.get_primary_hash() == first_event_hash
+        assert GroupHash.objects.get(group_id=new_event.group_id).group_id == group.id
+        assert new_event.group_id == group.id
+
+        # Now, if the group hash is deleted then the group will create a new group
+        assert len(GroupHash.objects.filter(group_id=group.id)) == 1
+        GroupHash.objects.get(group_id=group.id).delete()
```

Comment:
Once the group hash is fully deleted, everything will work, however, there's a race condition I'm investigating which recreates the group hash even after deletion.